src/AppRun.c

```diff
@@ -25,182 +25,298 @@ THE SOFTWARE.
+            * the next three characters to NULL CHAR. This acts as a sentry
+            * while parsing through all the substrings later.
+            * (We also confirm that the next two characters aren't already null.)
+            */
+            exe[n] = '\0';
+            exe[n+1] = '\0';
+            exe[n+2] = '\0';
+        } else if (exe[n] == '"' && exe[n-1] != '\\') {
+            /* If the current character is a non-escaped quote, toggle whether
+             * we are parsing inside a string.
+             */
+            in_quotes = !in_quotes;
+        } else if (line[n] == ' ' && !in_quotes) {
+            /* If the current character is a space, change it to a NULL CHAR,
+            * effectively splitting the string into multiple strings.
+            * This will allow us to iterate through the space-delimited args
+            * in the upcoming step.
+            */
+            exe[n] = '\0';
+        }
+    } while (exe[++n] != '\0');
+
+    // Create a pointer to each "substring" within `line`.
+    char* arg = exe;
+    // Counter for the number of arguments found.
+    int argcount = 0;
+
+    // Loop through each memory-adjacent string, counting them up.
+    while ((arg += (strlen(arg)+1)) && *arg != '\0') {
         argcount += 1;
+    }
 
-    // merge args
-    char*   outargptrs[argcount + argc + 1];
+    /* Merge the `Exec=` arguments with the `argv` arguments
+     * from the call to `main()`.
+     */
+
+    // Store the final list of arguments as an array of C-strings.
+    char* outargptrs[argcount + argc + 1];
+    // Store the name of the executable as the first argument.
     outargptrs[0] = exe;
-    int     outargindex = 1;
-    arg                 = exe;
-    int     argc_       = argc - 1;     // argv[0] is the filename
-    char**  argv_       = argv + 1;
+    // Keep track of which argument (index) we're storing.
+    int outargindex = 1;
+    /* Reset the `arg` pointer to the beginning of the sequence of
+     * memory-adjacent strings.
+     */
+    arg = exe;
+    // Skip the first argument, which is the filename, from the `main()` call.
+    int argc_ = argc - 1;
+    char** argv_ = argv + 1;
+    // Iterate through each argument from `Exec=`.
     while ((arg += (strlen(arg)+1)) && *arg) {
-        if (arg[0] == '%' || (arg[0] == '"' && arg[1] == '%')) {         // handle desktop file field codes
+        // If the current argument is a desktop file field code.
+        if (arg[0] == '%' || (arg[0] == '"' && arg[1] == '%')) {
+            // Get the actual code after the `%` or `"%`.
             char code = arg[arg[0] == '%' ? 1 : 2];
             switch(code) {
+                // For single file or URL args, merge in one of the `argv` args.
                 case 'f':
                 case 'u':
                     if (argc_ > 0) {
                         outargptrs[outargindex++] = *argv_++;
-                        argc_--;
+                        --argc_;
```

Comment:
Any specific reason you changed these calls? I mean, I don't care, really, either way is fine. Just curious.

Reply:
It's a performance habit. Although the compiler usually optimizes, it's good practice to use prefix instead of postfix, if the return order doesn't matter. It's one less instruction.